docs/src/main/paradox/cassandra.md

```diff
@@ -12,183 +18,117 @@ Unlogged batches are also supported.
+: @@snip [snip](/cassandra/src/test/java/docs/javadsl/CassandraSourceTest.java) { #cql }
 
-## Flow with passthrough
 
-Let's create a Cassandra Prepared statement with a query that we want to execute.
+If the statement requires specific settings, you may pass any @javadoc[com.datastax.oss.driver.api.core.cql.Statement](com.datastax.oss.driver.api.core.cql.Statement).
 
 Scala
-: @@snip [snip](/cassandra/src/test/scala/docs/scaladsl/CassandraSourceSpec.scala) { #prepared-statement-flow }
-
-Java
-: @@snip [snip](/cassandra/src/test/java/docs/javadsl/CassandraSourceTest.java) { #prepared-statement-flow }
-
-Now we need to create a 'statement binder', this is just a function to bind to the prepared statement. It can take in any type / data structure to fit your query values. Here we're just using one Integer, but it can just as easily be a (case) class.
-
-Scala
-: @@snip [snip](/cassandra/src/test/scala/docs/scaladsl/CassandraSourceSpec.scala) { #statement-binder-flow }
-
-Java
-: @@snip [snip](/cassandra/src/test/java/docs/javadsl/CassandraSourceTest.java) { #statement-binder-flow }
-
-We run the stream persisting the elements to C* and finally folding them using a ```Sink.fold```.
-
-Scala
-: @@snip [snip](/cassandra/src/test/scala/docs/scaladsl/CassandraSourceSpec.scala) { #run-flow }
+: @@snip [snip](/cassandra/src/test/scala/docs/scaladsl/CassandraSourceSpec.scala) { #statement }
 
 Java
-: @@snip [snip](/cassandra/src/test/java/docs/javadsl/CassandraSourceTest.java) { #run-flow }
-
-## Flow with passthrough and unlogged batching
+: @@snip [snip](/cassandra/src/test/java/docs/javadsl/CassandraSourceTest.java) { #statement }
 
-Use this when most of the elements in the stream share the same partition key. 
 
-Cassandra unlogged batches that share the same partition key will only
-resolve to one write internally in Cassandra, boosting write performance.
+Here we used a basic sink to complete the stream by collecting all of the stream elements to a collection. The power of streams comes from building larger data pipelines which leverage backpressure to ensure efficient flow control. Feel free to edit the example code and build @extref:[more advanced stream topologies](akka:stream/stream-introduction.html).
 
-**Be aware that this stage does not preserve the upstream order!**
 
-For this example we will define a class that model the data to be inserted
+## Updating Cassandra
 
-Scala
-: @@snip [snip](/cassandra/src/test/scala/docs/scaladsl/CassandraSourceSpec.scala) { #element-to-insert }
+@apidoc[CassandraFlow] provides factory methods to get Akka Streams flows to run CQL update statements. Alpakka Cassandra creates a @javadoc[PreparedStatement](com.datastax.oss.driver.api.core.cql.PreparedStatement) and for every stream element the `statementBinder` function binds the CQL placeholders to data.
```

Comment:
This makes it sound that only literal "update" statements are supported.  I assume you can insert or upsert as well.  Maybe this section could be rephrased to "Writing to Cassandra".